cmd/influx/config/config.go

```diff
@@ -95,34 +161,161 @@ func (svc LocalConfigsSVC) WriteConfigs(pp Configs) error {
 		if err == io.EOF {
 			break
 		}
-		b1.WriteString("# " + string(line) + "\n")
+		s.w.Write([]byte("# " + string(line) + "\n"))
 	}
-	return ioutil.WriteFile(svc.Path, b1.Bytes(), 0600)
+	return nil
 }
 
-// ParseConfigs decodes configs from io readers
-func ParseConfigs(r io.Reader) (Configs, error) {
-	p := make(Configs)
-	_, err := toml.DecodeReader(r, &p)
-	return p, err
+// ListConfigs decodes configs from io readers
+func (s baseRW) ListConfigs() (Configs, error) {
+	cfgs := make(Configs)
+	_, err := toml.DecodeReader(s.r, &cfgs)
+	for n, cfg := range cfgs {
+		cfg.Name = n
+		cfgs[n] = cfg
+	}
+	return cfgs, err
+}
+
+// CreateConfig create new config.
+func (svc localConfigsSVC) CreateConfig(cfg Config) (Config, error) {
+	if cfg.Name == "" {
+		return Config{}, &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Msg:  "config name is empty",
+		}
+	}
+	cfgs, err := svc.ListConfigs()
+	if err != nil {
+		return Config{}, err
+	}
+	if _, ok := cfgs[cfg.Name]; ok {
+		return Config{}, &influxdb.Error{
+			Code: influxdb.EConflict,
+			Msg:  fmt.Sprintf("config %q already exists", cfg.Name),
+		}
+	}
+	cfgs[cfg.Name] = cfg
+	if err := cfgs.Switch(cfg.Name); err != nil {
```

Comment:
why are we implicitly switching the new config to be the active one?

Reply:
it's been like that before.

Reply:
I understand that, I'm asking why thought? Why are we setting it active implicitly? IF that is the case we should remove the `active` flag from teh cli create command. It efffectively does nothing atm

Reply:
i already removed the active flag, to reflect the current behavior. For more discussion, let's move to https://github.com/influxdata/influxdb/issues/17534